src/github3/repos/comparison.py

```diff
@@ -65,17 +65,16 @@ def _update_attributes(self, compare):
         self.base_commit = commit.ShortCommit(compare["base_commit"], self)
         self.behind_by = compare["behind_by"]
         self.commits = compare["commits"]
-        if self.commits:
-            self.commits = [
-                commit.ShortCommit(com, self) for com in self.commits
-            ]
+        self.total_commits = compare["total_commits"]
+        self.commits = self._iter(
```

Comment:
I misunderstood what you'd suggested in your comment. This is hiding an API call so I won't accept this as is.

I would rename commits to original_commits and add a commits method here that returns the iter. That is aligned with the rest of this libraries design

Reply:
I have done this and modified the docstring for commits to original_commits, adding that original_commits is only the first few commits. Though about saying the first 250 but I guess that's dependent on github's implementation.